### ISSUE TITLE
No CMAKE_C_COMPILER could be found

### ISSUE THREAD

**Issue by OlafvdSpek** [microsoft/vcpkg#706]
Just installed 2017 RC on a Windows 8 system. Bootstrap and integration went fine, but no libraries can be build. Building a test project in the IDE works fine.

```
PS C:\vc\vcpkg> ./vcpkg install zlib
-- CURRENT_INSTALLED_DIR=C:/VC/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/VC/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/VC/vcpkg/packages/zlib_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/VC/vcpkg/buildtrees/zlib
-- CURRENT_PORT_DIR=C:/vc/vcpkg/ports/zlib/.
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Extracting source C:/VC/vcpkg/downloads/zlib1211.tar.gz
-- Extracting done
-- Configuring x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message):
    Command failed: C:/vc/vcpkg/downloads/cmake-3.8.0-rc1-win32-x86/bin/cmake.exe;C:/VC/vcpkg/buildtrees/zlib/src/zlib-1
.2.11;-DSKIP_INSTALL_FILES=ON;-DBUILD_SHARED_LIBS=ON;-DCMAKE_CXX_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP;-DCM
AKE_C_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /MP;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REG
ISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON;-DBoost_COMPILER=-vc140;-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /O
i /Gy /DNDEBUG /Zi;-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi;-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=/DEBUG /INCR
EMENTAL:NO /OPT:REF /OPT:ICF;-DCMAKE_EXE_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF;-G;Visual Studio
15 2017;-DCMAKE_VERBOSE_MAKEFILE=ON;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_TOOLCHAIN_FILE=C:/VC/vcpkg/triplets/x86-windows.c
make;-DCMAKE_PREFIX_PATH=C:/VC/vcpkg/installed/x86-windows;-DCMAKE_INSTALL_PREFIX=C:/VC/vcpkg/packages/zlib_x86-windows
    Working Directory: C:/VC/vcpkg/buildtrees/zlib/x86-windows-rel
    See logs for more information:
      C:\VC\vcpkg\buildtrees\zlib\config-x86-windows-rel-out.log
      C:\VC\vcpkg\buildtrees\zlib\config-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:95 (vcpkg_execute_required_process)
  ports/zlib/portfile.cmake:10 (vcpkg_configure_cmake)
  scripts/ports.cmake:73 (include)


Error: Building package zlib:x86-windows failed with: BUILD_FAILED
Please ensure sure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x86-windows
  Vcpkg version: 0.0.74-unknownhash
```

```
-- The C compiler identification is unknown
-- Configuring incomplete, errors occurred!
See also "C:/VC/vcpkg/buildtrees/zlib/x86-windows-rel/CMakeFiles/CMakeOutput.log".
See also "C:/VC/vcpkg/buildtrees/zlib/x86-windows-rel/CMakeFiles/CMakeError.log".
```

```
CMake Error at CMakeLists.txt:4 (project):
  No CMAKE_C_COMPILER could be found.
```

**Comment by ras0219-msft** [comment]
I'll see if I can reproduce this on a VM -- in the meantime, could you confirm that you can build a simple CMake project using the copy of CMake inside vcpkg?

`C:/vc/vcpkg/downloads/cmake-3.8.0-rc1-win32-x86/bin/cmake.exe`

**Comment by OlafvdSpek** [comment]
```
PS C:\vc\vcpkg-test> C:\VC\vcpkg\downloads\cmake-3.8.0-rc1-win32-x86\bin\cmake.exe .
-- Building for: Visual Studio 15 2017
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_C_COMPILER could be found.



CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.



-- Configuring incomplete, errors occurred!
See also "C:/VC/vcpkg-test/CMakeFiles/CMakeOutput.log".
See also "C:/VC/vcpkg-test/CMakeFiles/CMakeError.log".
PS C:\vc\vcpkg-test>
```

**Comment by ras0219-msft** [comment]
Now that VS2017 RTW is out, could you update your VS version and try again?

**Comment by OlafvdSpek** [comment]
RTW on W7 (not W8): CMakeError.log:

```
Compiling the C compiler identification source file "CMakeCCompilerId.c" failed.
Compiler:  
Build flags: 
Id flags:  

The output was:
1
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 9-3-2017 10:09:08.
Project "C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdC\CompilerIdC.vcxproj" on node 1 (default targets).
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Platforms\Win32\PlatformToolsets\v141\Toolset.targets(34,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdC\CompilerIdC.vcxproj]
Done Building Project "C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdC\CompilerIdC.vcxproj" (default targets) -- FAILED.

Build FAILED.

"C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdC\CompilerIdC.vcxproj" (default target) (1) ->
(Desktop_PlatformPrepareForBuild target) -> 
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Platforms\Win32\PlatformToolsets\v141\Toolset.targets(34,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdC\CompilerIdC.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.95


Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" failed.
Compiler:  
Build flags: 
Id flags:  

The output was:
1
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 9-3-2017 10:09:10.
Project "C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj" on node 1 (default targets).
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Platforms\Win32\PlatformToolsets\v141\Toolset.targets(34,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj]
Done Building Project "C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj" (default targets) -- FAILED.

Build FAILED.

"C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj" (default target) (1) ->
(Desktop_PlatformPrepareForBuild target) -> 
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Platforms\Win32\PlatformToolsets\v141\Toolset.targets(34,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.93
```

OpenSSL installed just fine though:
Package openssl:x86-windows is installed

**Comment by derkestrel** [comment]
I ran into the same problem. OpenSSL worked, but packages depending on zlib fail. I think the "cmake not found" is a red herring.

 Digging deeper into the logs it seems the linking of zlib fails.

from C:\Users\User\Sources\vcpkg\buildtrees\zlib\x86-windows-rel\CMakeFiles\CMakeError.log
Section Link:
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x86\link.exe /ERRORREPORT:QUEUE /OUT:".\CompilerIdC.exe" /INCREMENTAL:NO /NOLOGO /LIBPATH:"C:\Users\user\Sources\vcpkg\installed\x86-windows\debug\lib" /LIBPATH:"C:\Users\user\Sources\vcpkg\installed\x86-windows\debug\lib\manual-link" kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib "C:\Users\user\Sources\vcpkg\installed\x86-windows\debug\lib*.lib" /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /PDB:".\CompilerIdC.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:".\CompilerIdC.lib" /MACHINE:X86 /SAFESEH Debug\CMakeCCompilerId.obj
LINK : fatal error LNK1104: cannot open file 'ucrtd.lib'

Is this the same for you?

**Comment by OlafvdSpek** [comment]
No, zlib works fine for me. I'm stuck on 

```
error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\VC\vcpkg-test\CMakeFiles\3.8.0-rc1\CompilerIdCXX\CompilerIdCXX.vcxproj]
```

VS only installs Windows SDK 10 by default.

**Comment by derkestrel** [comment]
Did you try installing [(https://developer.microsoft.com/en-us/windows/downloads/windows-8-1-sdk)] ?

**Comment by OlafvdSpek** [comment]
No. It should use the Windows SDK 10.

**Comment by ras0219-msft** [comment]
Ok, we've pushed a change that should fix this issue (selects the win10 sdk). Could you update and try again?

**Comment by OlafvdSpek** [comment]
./vcpkg install aws-sdk-cpp

```
CMake Error at C:/VC/vcpkg/downloads/cmake-3.8.0-rc1-win32-x86/share/cmake-3.8/Modules/CMakeTestCXXCompiler.cmake:44 (message):
  The C++ compiler "C:/Program Files (x86)/Microsoft Visual
  Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe" is
  not able to compile a simple test program.

  It fails with the following output:

   Change Dir: C:/VC/vcpkg/buildtrees/aws-sdk-cpp/x86-windows-rel/CMakeFiles/CMakeTmp

  

  Run Build Command:"C:/VC/vcpkg/downloads/tools/ninja/ninja-1.7.2/ninja.exe"
  "cmTC_87a79"

  [1/2] Building CXX object
  CMakeFiles\cmTC_87a79.dir\testCXXCompiler.cxx.obj

  [2/2] Linking CXX executable cmTC_87a79

  FAILED: cmTC_87a79 

  cmd.exe /C "cd .  &&
  C:\VC\vcpkg\downloads\cmake-3.8.0-rc1-win32-x86\bin\cmake.exe -E
  vs_link_exe --intdir=CMakeFiles\cmTC_87a79.dir --manifests --
  C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1410~1.250\bin\HostX86\x86\link.exe
  /nologo CMakeFiles\cmTC_87a79.dir\testCXXCompiler.cxx.obj /out:cmTC_87a79
  /implib: /pdb:cmTC_87a79.pdb /version:0.0 /machine:X86 /debug /INCREMENTAL
  /subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib
  shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib &&
  cmd.exe /C "cd /D
  C:\VC\vcpkg\buildtrees\aws-sdk-cpp\x86-windows-rel\CMakeFiles\CMakeTmp &&
  powershell -noprofile -executionpolicy UnRestricted -file
  C:/VC/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary
  C:/VC/vcpkg/buildtrees/aws-sdk-cpp/x86-windows-rel/CMakeFiles/CMakeTmp/cmTC_87a79
  -installedDir C:/VC/vcpkg/installed/x86-windows/debug/bin -OutVariable
  out""

  LINK : fatal error LNK1146: no argument specified with option '/implib:'


  LINK Pass 1 failed.  with 1146

  ninja: build stopped: subcommand failed.

  

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:99 (project)



```

**Comment by jbhelm** [comment]
Just wanted to add I was also getting the erroneous "No CMAKE_C_COMPILER could be found" error when attempting to build zlib. In my case, the error in CMakeError.log was:
```
  powershell.exe -ExecutionPolicy Unrestricted -noprofile -File "C:\dev\vcpkg\scripts\buildsystems\msbuild\applocal.ps1" "C:\dev\vcpkg\buildtrees\zlib\x86-windows-rel\CMakeFiles\3.8.0-rc1\CompilerIdC\.\CompilerIdC.exe" "C:\dev\vcpkg\installed\x86-windows\debug\bin" "Debug\CompilerIdC.tlog\CompilerIdC.write.1u.tlog"
  AuthorizationManager check failed.
      + CategoryInfo          : SecurityError: (:) [], ParentContainsErrorRecordException
      + FullyQualifiedErrorId : UnauthorizedAccess
```

It's caused by the fact that the file "scripts\buildsystems\msbuild\applocal.ps1" was considered "blocked" by the OS, which results in PowerShell not being able to execute the script. Selecting Properties in Windows Explorer on the file and checking the "Unblock" (This file came from another computer and might be blocked to help protect this computer) option resolved the issue for me.

**Comment by ras0219-msft** [comment]
I've just pushed eb9b85c49 which should help with this -- we had a few remaining uses of `Unrestricted` in our sources which would trigger failure due to Mark Of The Web.

**Comment by nlappas** [comment]
Any updates? I tried to open a folder that contained a CMakelists.txt and got the same error in Windows 7. Meanwhile If I try the cmake 3.8.1 from cmd, it finds properly the VS C compiler.

(BTW it works in Windows 10)

**Comment by QianyuyuGitHub** [comment]
Just want this to be clear, I also got the same situation: the Cmake error no C compiler. After I modify the **VS2017** by installing the **win10 SDK**, it works out with me! So if anyone got the same problem, please check your VS first! And appreciate all the people under this issue, I finally get through it by your helps!

**Comment by how0723** [comment]
maybe your system languages do not contain english ,,,or your languages do not contain english

**Comment by PhoebeHui** [comment]
I think it's safe to close this thread, please report a new issue and attach failures if you encounter similar issue.
